source/manual/github-unavailable.html.md

```diff
@@ -23,9 +23,92 @@ We [mirror all GitHub repositories](repository-mirroring.html) tagged with `govu
+GOV.UK AWS admin users can give access to developers who need to make changes to the code before deployment.
+
+1. In the root of the local repo, run the following commands to install the AWS
+   credential helper and add CodeCommit as a remote:
+
+   ```
+   git config credential.helper '!aws codecommit credential-helper $@'
+   git config credential.UseHttpPath true
+   git remote add aws https://git-codecommit.eu-west-2.amazonaws.com/v1/repos/<app>
+   ```
+
+1. [Get some AWS credentials](/manual/deploying-terraform.html#2-get-your-credentials)
+   for the `govuk-tools` AWS account
+
+1. Export the access key ID, secret access key and session token from the last step,
+   for example:
+
+   ```
+   export AWS_ACCESS_KEY_ID=...
+   export AWS_SECRET_ACCESS_KEY=...
+   export AWS_SESSION_TOKEN=...
+   ```
+
+1. Fetch the AWS upstream by running `git fetch aws`
+
+1. Checkout a new branch on the upstream by running `git checkout -b aws/my-super-secret-fix`
+
+1. Make and commit your changes to this branch, and make sure all tests run successfully
+   locally (since CodeCommit does not run tests)
+
+1. Push your changes to CodeCommit by running `git push`
+
+1. Tag your changes by running `git tag release_XYZ`, where XYZ is one more that the latest
+   release tag for the application you're working on, as reported by the Release app
+
+1. Push your new tag to CodeCommit by running `git push aws release_XYZ`
+
+#### Deploying the code change
+
+1. Review the pull request on AWS CodeCommit.
+
+1. Create a release tag manually in git. This should follow the standard format
+   `release_X`. Tag the branch directly instead of merging it.
+
+1. Don't use the release app. Go directly to the `Deploy_App` Jenkins job, and
+   check `DEPLOY_FROM_AWS_CODECOMMIT`.
+
+#### After deploying the change
+
+1. Push the branch and tag to GitHub.
+
+1. Merge the branch into master.
+
+1. Record the missing deployment in the Release app.
+
+#### Troubleshooting
+
+If running any `git` commands against CodeCommit returns you a 403, you probably
+have expired credentials stored in your MacOS keychain from a previous attempt.
+Apparently MacOS stores these the first time you use it and subsequently tries
+to use them again despite you setting new AWS credentials.
+
+To fix this:
+
+1. Open Keychain Access (use cmd-space to search for it).
+
+1. Select "Passwords" from the "Category" on the left.
+
+1. Search for `git-codecommit`.
+
+1. Right click on the item and select "Get Info".
+
+1. Click "Access Control" on the modal that pops up.
+
+1. Select "git-credential-osxkeychain" from the list.
+
+1. Hit the minus key.
+
+1. Try your terminal commands again.
+
+1. If you are prompted to add the item to keychain, deny.
+
+There is more information in Step 3 in the [AWS
```

Comment:
I'm a little confused here: should this relate to the export access key stuff from earlier?